src/Tribe/iCal.php

```diff
@@ -588,125 +589,137 @@ protected function get_body( $posts = [] ) {
 		$events      = '';
 
 		foreach ( $posts as $event_post ) {
-			// add fields to iCal output
-			$item = [];
-
-			$full_format = 'Ymd\THis';
-			$utc_format  = 'Ymd\THis\Z';
-			$all_day     = ( 'yes' === get_post_meta( $event_post->ID, '_EventAllDay', true ) );
-			$time = (object) [
-				'start' => tribe_get_start_date( $event_post->ID, false, 'U' ),
-				'end' => tribe_get_end_date( $event_post->ID, false, 'U' ),
-				'modified' => Tribe__Date_Utils::wp_strtotime( $event_post->post_modified ),
-				'created' => Tribe__Date_Utils::wp_strtotime( $event_post->post_date ),
-			];
 
-			$type   = 'DATE-TIME';
-			$format = $full_format;
+			$item = $this->get_ical_output_for_an_event( $event_post, $tec );
 
-			if ( $all_day ) {
-				$type   = 'DATE';
-				$format = 'Ymd';
-			}
+			$events .= "BEGIN:VEVENT\r\n" . implode( "\r\n", $item ) . "\r\nEND:VEVENT\r\n";
+		}
 
-			$tzoned = (object) [
-				'start' => date( $format, $time->start ),
-				'end' => date( $format, $time->end ),
-				'modified' => date( $utc_format, $time->modified ),
-				'created' => date( $utc_format, $time->created ),
-			];
+		return $events;
+	}
 
-			$dtstart = $tzoned->start;
-			$dtend   = $tzoned->end;
+	/**
+	 * Get the iCal Output for the provided event object.
+	 *
+	 * @since TBD
+	 *
+	 * @param \WP_Post             $event_post The event post object.
+	 * @param \Tribe__Events__Main $tec        An instance of the main TEC Class.
+	 *
+	 * @return array  An array of iCal output fields.
+	 */
+	public function get_ical_output_for_an_event( $event_post, Tribe__Events__Main $tec ) {
 
-			if ( 'DATE' === $type ) {
-				// For all day events dtend should always be +1 day.
-				if ( $all_day ) {
-					$dtend = date( $format, strtotime( '+1 day', strtotime( $dtend ) ) );
-				}
+		// Add fields to iCal output.
+		$item = [];
 
-				$item[] = 'DTSTART;VALUE=' . $type . ':' . $dtstart;
-				$item[] = 'DTEND;VALUE=' . $type . ':' . $dtend;
-			} else {
-				// Are we using the sitewide timezone or the local event timezone?
-				$timezone_name = $this->get_timezone( $event_post );
-				$timezone = Tribe__Events__Timezones::build_timezone_object( $timezone_name );
+		$full_format = 'Ymd\THis';
+		$utc_format  = 'Ymd\THis\Z';
+		$all_day     = ( 'yes' === get_post_meta( $event_post->ID, '_EventAllDay', true ) );
+		$time        = (object) [
+			'start'    => tribe_get_start_date( $event_post->ID, false, 'U' ),
+			'end'      => tribe_get_end_date( $event_post->ID, false, 'U' ),
```

Comment:
These too.

Reply:
I tried to use a different method on these, but I could not get the same timestamp it current produces. So I left it for now.